lib/solidus_support/engine_extensions.rb

```diff
@@ -46,14 +46,38 @@ def load_solidus_subscribers_from(path)
         end
       end
 
-      # Loads decorator files.
+      # Loads decorators.
       #
       # This is needed since they are never explicitly referenced in the application code and
-      # won't be loaded by default. We need them to be executed regardless in order to decorate
-      # existing classes.
-      def load_solidus_decorators_from(path)
-        path.glob('**/*.rb') do |decorator_path|
-          require_dependency(decorator_path)
+      # won't be loaded by default. We need them to be executed whenever the decorated class is reloaded.
+      def load_solidus_decorators_from(base_path)
+        # Matches e.g. "Spree::Order.prepend"
+        decorated_class_pattern = /(?<decorated_class>[A-Z][a-zA-Z:]+)(\.prepend[\s(])/
+        # This will be Zeitwerk.
+        autoloader = Rails.autoloaders.main
+
+        base_path.glob('**/*.rb') do |path|
+          # Match all the classes that are prepended in the file
+          matches = File.open(path).read.scan(decorated_class_pattern).flatten
```

Comment:
Would this work as well?

```suggestion
          matches = File.read(path).scan(decorated_class_pattern).flatten
```

Reply:
Yes it would! Changing.